### ISSUE TITLE
Request flatpak 

### ISSUE THREAD

**Issue by Danathar** [ollama/ollama#2288]
I'd like to request a flatpak. It's easier to install, easy to sandbox, safer than piping a script into bash and cross platform! You can then submit the flatpak to flathub, get the verified icon and things would be awesome! ;)

thanks!

**Comment by terminal-cs** [comment]
+1 plz do this

**Comment by rugk** [comment]
I agree! It would be nice if you could publish this as a [flatpak](https://flatpak.org/) on [flathub](https://flathub.org/) e.g.

Flatpaks are a new software distribution mechanism for Linux distros, can thus installed on any distro and are easy to update. They are easy to install _and_ update and work on all Linux distros.
Also, if you publish it on _FlatHub_ you may grow your user base given many distros include that and it is a common software source, so your app can be discovered more easily. 

Here is [how to get started](http://docs.flatpak.org/en/latest/getting-started.html).

**Comment by RuofengX** [comment]
Agree. I am in China, the international connect is not good sometime. The process of install the ollame cli will benefit a lot from flatpak mirrors.

**Comment by tsilvs** [comment]
There is an [OpenSUSE RPM of Ollama](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/ollama-0.1.38-2.1.x86_64.rpm.html) ([in "Factory" builds repo](https://build.opensuse.org/package/show/openSUSE:Factory/ollama)), maintained by [dimstar](https://build.opensuse.org/users/dimstar_suse) @DimStar77

It's ~10 times smaller than the official builds. Flatpak build could probably be optimized in the same way.

**Comment by tsilvs** [comment]
Also without a Flatpak or a native package release (e.g. `.rpm`) installation process is not very compatible with immutable root distros (e.g. Fedora Silverblue under ostree).

**Comment by richardstevenhack** [comment]
I second this. I want to run Ollama on openSUSE which at the moment it doesn't (due to my inadequate GPU) and having to install and run it as root due to it being installed in the wrong location. A flatpak or AppImage would likely solve the latter.

**Comment by wiiznokes** [comment]
Flatpak is not meant for cli, so i will never be accepted in flathub. There is a [snap](https://snapcraft.io/ollama), but it doesn't seems to be very maintained, and my nvidia card was not used

**Comment by richardstevenhack** [comment]
I found an openSUSE package which does install properly and runs version 0.1.40, so only 4 release behind.  :-)
https://software.opensuse.org/package/ollama

**Comment by Danathar** [comment]
> Flatpak is not meant for cli, so it will never be accepted in flathub. There is a [snap](https://snapcraft.io/ollama), but it doesn't seems to be very maintained, and my nvidia card was not used

That's actually a very good point. I had completely forgotten about the fact that flatpaks are not really for command-line or server-side software. Although I always go for flatpaks when available, if a snap is the way to go, then yes, I could get behind that.

**Comment by tsilvs** [comment]
> > Flatpak is not meant for cli
> flatpaks are not really for command-line

Flatpaks can be called as `flatpak run org.example.app --opt1 --opt2 --etc` and even aliased as `alias app="flatpak run org.example.app"`

**Comment by Anvil5465** [comment]
fyi in case this is helpful to anyone following this--there is an unofficial GUI for ollama with a flatpak package called [Alpaca](https://github.com/Jeffser/Alpaca)

**Comment by richardstevenhack** [comment]
Thanks for the link to Alpaca. Just installed it and testing. It's limited, but does the job of a simple Ollama interface. 

For anyone considering it, note that the default Ollama port was changed to 11435 instead of 11434. I had to change it back in the config file for it to connect to my running Ollama. 

Also I think if you pull a model using its interface, it will store the model in ~/.var/app/com.jeffser.Alpaca/data/.ollama/models/ whereas on my openSUSE Tumbleweed system, Ollama stores its models in /var/lib/ollama/.ollama/models/. So if you pull models using Alpaca, you'll end up with two models in two difference places or you'll have to copy them from one to the other.

**Comment by wiiznokes** [comment]
> > > Flatpak is not meant for cli
> > 
> > 
> > flatpaks are not really for command-line
> 
> Flatpaks can be called as `flatpak run org.example.app --opt1 --opt2 --etc` from terminal and even aliased as `alias app="flatpak run org.example.app"`

That is true, i apologize. The [doc](https://docs.flathub.org/docs/for-app-authors/requirements/#desktop-integration) say

> 
> Flathub is primarily focused on graphical desktop applications and they have the following expectations to ensure effective desktop integration.
> 
> Do note that CLI applications do not require a .desktop file but TUI applications should have a .desktop file with Terminal=true.

So it should be ok to package it with flatpak. That being said, running `flatpak run org.example.app` each time is kinda disappointing (i know there are script for that, but it require extra steps. Also, i'm not sure if flatpak can integrate a system service but i may be wrong

**Comment by rugk** [comment]
FYI another animal-named GUI for AI models, that is (now) on flathub is BTW this one: https://flathub.org/apps/com.jeffser.Alpaca

**Comment by otuva** [comment]
I do have some experience with flatpaks and their maintenance. I'd like to contribute.
Is this also wanted by repo owners?

**Comment by ali1234** [comment]
> Flatpak is not meant for cli, so it will never be accepted in flathub. There is a [snap](https://snapcraft.io/ollama), but it doesn't seems to be very maintained, and my nvidia card was not used

It's a year later now and this snap seems to be quite well maintained: at time of writing the stable channel is on 0.9.0 and the beta channel is on 0.9.2 (tagged 11 days ago, and snapped 6 days ago, a lag of 5 days). 0.9.3 (tagged 3 days ago) has not been snapped yet, but I expect it will be sooner rather than later. This snap does use nvidia for me and generally seems to work exactly as you would expect, at least on Ubuntu.

There is a snap for open-webui from the same maintainer, but that only has a beta channel and it is over a year old. I can't speak for how well it works as I have not tried to use it.

**Comment by ser** [comment]
> FYI another animal-named GUI for AI models, that is (now) on flathub is BTW this one: https://flathub.org/apps/com.jeffser.Alpaca

from inside this package you can install ollama only, `flatpak install flathub com.jeffser.Alpaca.Plugins.Ollama`

**Comment by Danathar** [comment]
I've kinda changed my tune on this since deploying it as a quadman podlet, though I'd still like to see a more platform agnostic way of deploying it for people that don't want to use podman or docker.

**Comment by richardstevenhack** [comment]
I now use MSTY Studio - https://msty.ai/ -  which embeds a copy of the Ollama server in its own system setup. 

The only downside is when Ollama updates its server, you have to download it manually and move it into MSTY's config directory. They don't automatically update the server themselves. A minor issue.

MSTY itself has enormous capabilities, including serving the embedded Ollama server to other third-party utilities. Unless you're fixated on the command line, a GUI like this is the way to go to run local and remote LLMs.
